docs/generator.py

```diff
@@ -0,0 +1,43 @@
+from pytablewriter import MarkdownTableWriter
+from pytablewriter.style import Style
+
+import jupyterhub.metrics
+
+
+class Generator:
+    @classmethod
+    def create_writer(cls, table_name, headers, values):
+        writer = MarkdownTableWriter()
+        writer.table_name = table_name
+        writer.headers = headers
+        writer.value_matrix = values
+        writer.margin = 1
+        [writer.set_style(header, Style(align="center")) for header in headers]
+        return writer
+
+    def _parse_metrics(self):
+        table_rows = []
+        for name in dir(jupyterhub.metrics):
+            obj = getattr(jupyterhub.metrics, name)
+            if obj.__class__.__module__.startswith('prometheus_client.'):
+                for metric in obj.describe():
+                    table_rows.append([metric.type, metric.name, metric.documentation])
+        return table_rows
+
+    def prometheus_metrics(self):
+        filename = "./source/monitoring/metrics.md"
```

Comment:
Let's make sure to use a path relative to `__file__` or an input, since this will result in a different generated file depending on the CWD when the script is called.

Let's also put it in a `source/monitoring/gen/metrics.md` file to be extra clear that it's a generated file and the whole directory can go in .gitignore